src/library/scala/math/Ordering.scala

```diff
@@ -109,9 +109,27 @@ trait Ordering[T] extends Comparator[T] with PartialOrdering[T] with Serializabl
   /** Return `x` if `x` <= `y`, otherwise `y`. */
   def min[U <: T](x: U, y: U): U = if (lteq(x, y)) x else y
 
-  /** Return the opposite ordering of this one. */
+  /** Return the opposite ordering of this one.
+    *
+    * Implementations overriding this method MUST override [[reverseOf]]
+    * as well if they change the behavior at all (however, caching does
+    * not require overriding it).
+    */
   override def reverse: Ordering[T] = new Ordering.Reverse[T](this)
 
+  /** Returns whether or not the other ordering is the opposite
+    * ordering of this one.
+    *
+    * Equivalent to `other == this.reverse`.
+    *
+    * Implementations should only override this method if they are overriding
+    * [[reverse]] as well.
+    */
+  def reverseOf(other: Ordering[_]): Boolean = other match {
```

Comment:
I'd call the method `isReverseOf`, would that work for you too?

Reply:
sounds good to me! I had trouble deciding between the two names